package/opensuse-build-service-helper.rb

```diff
@@ -0,0 +1,164 @@
+# Copyright(C) 2023 Kentaro Hayashi <hayashi@clear-code.com>
+#
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this library; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+
+require "pathname"
+
+module OpenSUSEBuildServiceHelper
+
+  def define_obsbuild_tasks
+    ensure_osc_configuration
+    define_apt_obsbuild_task
+    define_yum_obsbuild_task
+  end
+
+  private
+
+  def obs_project_name
+    # e.g. isv:milter-manager
+    ENV["OBS_PROJECT"] ? ENV["OBS_PROJECT"] : "isv:#{@package}"
+  end
+
+  def obs_package_dir(package_name)
+    File.join(obs_project_name, package_name)
+  end
+
+  def oscrc_path
+    Pathname.new("~/.config/osc/oscrc").expand_path
+  end
+
+  def obs_user
+    obs_user = nil
+    if oscrc_path.exist?
+      oscrc_content = oscrc_path.read
+    else
+      oscrc_content = ""
+    end
+    oscrc_content.each_line do |line|
+      if line.start_with?("user=")
+        obs_user = line.split('=', 2).last.chomp
+      end
+    end
+    obs_user ? obs_user : ENV["USER"]
+  end
+
+  def srpm_sources_dir
+    Pathname("#{Dir.pwd}/home/#{ENV['USER']}/rpmbuild/SOURCES")
+  end
+
+  def srpm_spec_dir
+    Pathname("#{Dir.pwd}/home/#{ENV['USER']}/rpmbuild/SPECS")
+  end
+
+  def ensure_osc_configuration
+    unless oscrc_path.exist?
+      raise "can't detect #{oscrc_path}"
+    end
+  end
+
+  def apply_obs_pkg_metadata(project_name, package_name)
```

Comment:
How about using `obs_` prefix?

```suggestion
  def obs_apply_pkg_metadata(project_name, package_name)
```